lib/src/editor/ui/editor.dart

```diff
@@ -59,32 +68,75 @@ class Editor extends StatelessWidget {
                         },
                       ),
                     ),
-                    Consumer<ZoomProvider>(
+                    Consumer<TabProvider>(
                       builder: (context, model, child) {
-                        TransformationController _transformation =
-                            TransformationController();
-                        _transformation.value = Matrix4.identity()
-                          ..scale(model.zoom);
-                        return context.watch<DevicePreviewProvider>().show
-                            ? DevicePreview(
-                                builder: (context) {
-                                  return InteractiveViewer(
-                                    panEnabled: true,
-                                    boundaryMargin:
-                                        EdgeInsets.all(double.infinity),
-                                    child: component ?? const SizedBox.shrink(),
-                                    transformationController: _transformation,
-                                  );
+                        List<ComponentState> state = recursiveRetrievalOfStates(
+                            context.read<List<Category>>());
+
+                        return model.tab == editor.FlutterBookTab.canvas
```

Comment:
⛏️  -  Using the `? : ` syntax and being nested takes a bit to follow along. For longer block of code like this we might want to consider different syntax to make it easier to read.

Reply:
agreed.

I'd make this into 3-4 smaller private widgets and then the ternaries would not need to be nested in a single widget anymore

i.e

```
_Canvas
_Docs
_DevicePreviewCanvas
_InteractiveViewerCanvas 

```

Reply:
thanks !

Reply:
👍 Much easier to follow along and understand